packages/mux-uploader/src/index.ts

```diff
@@ -0,0 +1,419 @@
+    const uploaderHtml = template.content.cloneNode(true);
+    shadow.appendChild(uploaderHtml);
+
+    this.hiddenFileInput = this.shadowRoot?.querySelector('input[type="file"]');
+    this.filePickerButton = this.shadowRoot?.querySelector('button');
+    this.svgCircle = this.shadowRoot?.querySelector('circle');
+    this.progressBar = this.shadowRoot?.getElementById('progress-bar');
+    this.uploadPercentage = this.shadowRoot?.getElementById('upload-status');
+    this.statusMessage = this.shadowRoot?.getElementById('status-message');
+    this.retryMessage = this.shadowRoot?.getElementById('retry-message');
+
+    this._dropHandler = this.handleUpload.bind(this);
+  }
+
+  connectedCallback() {
+    this.setDefaultType();
+    this.setupFilePickerButton();
+    this.setupRetry();
+    this.setupDropHandler();
+
+    this.setAttribute('role', 'progressbar');
+    this.setAttribute('aria-label', 'progress bar');
+    this.setAttribute('aria-live', 'polite');
+
+    // TO-DO: Might want to standardize if we prefer to have users disable or enable things. (TD).
+    // Edge case: User wants to use the uploader without drag.
+    // Because we slot a default mux-uploader-drop if they don't slot one,
+    // in order to disable drop, the user has to pass disableDrop to mux-uploader and mux-uploader-drop
+    // must apply it in order to disable the drop. It currently requires you to explicit pass "true".
+    if (this.getAttribute('disableDrop')) {
+      const muxUploaderDrop = this.shadowRoot?.querySelector('mux-uploader-drop');
+      muxUploaderDrop?.setAttribute('disableDrop', '');
+    }
+  }
+
+  disconnectedCallback() {
+    //@ts-ignore
+    this.removeEventListener('mux-drop', this._dropHandler, false);
+  }
+
+  get radius() {
+    return Number(this.svgCircle?.getAttribute('r'));
+  }
+
+  get circumference() {
+    return this.radius * 2 * Math.PI;
+  }
+
+  setDefaultType() {
+    const currentType = this.getAttribute('type');
+
+    if (!currentType) {
+      this.setAttribute('type', TYPES.BAR);
+    }
+
+    if (currentType === TYPES.RADIAL) {
+      if (this.svgCircle) {
+        // strokeDasharray is the size of dashes used to draw the circle with the size of gaps in between.
+        // If the dash number is the same as the gap number, no gap is visible: a full circle.
+        // strokeDashoffset defines where along our circle the dashes (in our case, a dash as long as the
+        // circumference of our circle) begins. The larger the offset, the farther into the circle you're
+        // starting the "dash". In the beginning, offset is the same as the circumference. Meaning, the visible
+        // dash starts at the end so we don't see the full circle. Instead we see a gap the size of the circle.
+        // When the percentage is 100%, offset is 0 meaning the dash starts at the beginning so we can see the circle. (TD).
+
+        this.svgCircle.style.strokeDasharray = `${this.circumference} ${this.circumference}`;
```

Comment:
followup(non-blocking): Following the code thread here, we may be able to do this entirely through css (🤞)